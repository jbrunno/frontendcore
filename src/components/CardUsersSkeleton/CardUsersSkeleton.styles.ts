import { styled, css } from '@mui/material/styles';
import Card from '../Card/Card';
import Skeleton from '../Skeleton/Skeleton';

export const CardStyled = styled(Card)(
  ({ theme }) => css`
    display: grid;
    grid-template-columns: auto 1fr auto;
    align-items: center;
    grid-gap: ${theme.fn.spacing(16)};
    padding: ${theme.fn.spacing(16)};
  `,
);

export const SkeletonStyled = styled(Skeleton)`
  border-radius: 8px;
`;
