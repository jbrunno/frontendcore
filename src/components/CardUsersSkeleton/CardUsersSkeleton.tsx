import React from 'react';
import Box from '../Box/Box';
import { CardProps } from '../Card/Card';
import Skeleton from '../Skeleton/Skeleton';
import { CardStyled, SkeletonStyled } from './CardUsersSkeleton.styles';

type CardUsersSkeletonProps = CardProps;

export function CardUsersSkeleton(props: CardUsersSkeletonProps) {
  return (
    <CardStyled {...props}>
      <SkeletonStyled variant="rectangular" width={64} height={64} />
      <Box>
        <Skeleton variant="text" height={24} />
        <Skeleton variant="text" height={15} />
      </Box>
    </CardStyled>
  );
}

export default CardUsersSkeleton;
