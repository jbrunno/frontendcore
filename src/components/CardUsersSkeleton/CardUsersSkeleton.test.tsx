import React from 'react';
import { render } from '@test-utils/component-testing';
import { CardUsersSkeleton } from './CardUsersSkeleton';

test('renders the CardUsersSkeleton', () => {
  const { container } = render(<CardUsersSkeleton />);
  expect(container).toMatchSnapshot();
});
