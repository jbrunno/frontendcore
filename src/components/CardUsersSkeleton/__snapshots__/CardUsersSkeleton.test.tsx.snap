// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CardUsersSkeleton 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-2 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  background-color: #fff;
  display: grid;
  grid-template-columns: auto 1fr auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  padding: 1rem;
}

.emotion-4 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: 1.2em;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
  border-radius: 8px;
}

.emotion-6 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: auto;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px/6.7px;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
}

.emotion-6:empty:before {
  content: "\\00a0";
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0 emotion-1 emotion-2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse emotion-3 emotion-4"
      style="width: 64px; height: 64px;"
    />
    <div
      class="MuiBox-root emotion-5"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-6"
        style="height: 24px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-6"
        style="height: 15px;"
      />
    </div>
  </div>
</div>
`;
